From: Jo-Philipp Wich <jo@mein.io>
Date: Thu, 23 Aug 2018 19:08:58 +0200
Subject: grub2: rebase patches

Patch 300-CVE-2015-8370.patch was added without proper rebasing on the
version used by OpenWrt, make it apply and refresh the patch to fix
compilation.

Fixes: 7e73e9128f ("grub2: Fix CVE-2015-8370")
Signed-off-by: Jo-Philipp Wich <jo@mein.io>
(cherry picked from commit 9ffbe84ea49fc643f41bfdf687de99aee17c9154)

diff --git a/package/boot/grub2/patches/300-CVE-2015-8370.patch b/package/boot/grub2/patches/300-CVE-2015-8370.patch
index 11bc4340c08d559ec00c8303740d4512f3cb1a12..eb7919639a895e1da9dc42410c9a07a31d22a073 100644
--- a/package/boot/grub2/patches/300-CVE-2015-8370.patch
+++ b/package/boot/grub2/patches/300-CVE-2015-8370.patch
@@ -16,11 +16,9 @@ Signed-off-by: Ismael Ripoll-Ripoll <iripoll@disca.upv.es>
  grub-core/normal/auth.c | 2 +-
  2 files changed, 2 insertions(+), 2 deletions(-)
 
-diff --git a/grub-core/lib/crypto.c b/grub-core/lib/crypto.c
-index ca334d5a40e..e6c78d16d39 100644
 --- a/grub-core/lib/crypto.c
 +++ b/grub-core/lib/crypto.c
-@@ -468,7 +468,7 @@ grub_password_get (char buf[], unsigned buf_size)
+@@ -468,7 +468,7 @@ grub_password_get (char buf[], unsigned
  	  break;
  	}
  
@@ -29,16 +27,14 @@ index ca334d5a40e..e6c78d16d39 100644
  	{
  	  if (cur_len)
  	    cur_len--;
-diff --git a/grub-core/normal/auth.c b/grub-core/normal/auth.c
-index 6be678c0de1..c35ce972473 100644
 --- a/grub-core/normal/auth.c
 +++ b/grub-core/normal/auth.c
-@@ -172,7 +172,7 @@ grub_username_get (char buf[], unsigned buf_size)
+@@ -172,7 +172,7 @@ grub_username_get (char buf[], unsigned
  	  break;
  	}
  
--      if (key == GRUB_TERM_BACKSPACE)
-+      if (key == GRUB_TERM_BACKSPACE && cur_len)
+-      if (key == '\b')
++      if (key == '\b' && cur_len)
  	{
  	  if (cur_len)
  	    {
